mission3_new/src/routers/articleRouter.js

```diff
@@ -0,0 +1,154 @@
+  } else {
+    orderBy = order;
+  }
+
+  const whereClause = keyword
+    ? {
+        OR: [
+          { content: { contains: keyword, mode: 'insensitive' } },
+          { title: { contains: keyword, mode: 'insensitive' } },
+        ],
+      }
+    : {};
+
+  const article = await prisma.article.findMany({
+    where: whereClause,
+    orderBy: {
+      createdAt: orderBy,
+    },
+    skip: parseInt((page - 1) * pageSize),
+    take: parseInt(pageSize),
+
+    select: {
+      id: true,
+      title: true,
+      content: true,
+      createdAt: true,
+    },
+  });
+  res.status(200).json(article);
+}));
+////////////////////////////////////////////////////////////////
+router.get('/:aid', asyncHandler(async (req, res) => {
+  const { aid } = req.params;
+  const aaid = parseInt(aid);
+  const article = await prisma.article.findUnique({
+    where: { id: aaid },
+    select: {
+      id: true,
+      title: true,
+      content: true,
+      createdAt: true,
+    },
+  });
+  res.status(200).json(article);
+}));
+//////////////////////////////////////////////////////////////성공
+
+router.post('/', asyncHandler(async (req, res) => {
+  assert(req.body, CreateArticle);
+  const article = await prisma.article.create({
+    data: req.body,
+  });
+  res.status(201).json(article);
+}));
+
+//////////////////////////////////////////////////////////////
+
+router.patch('/:aid', asyncHandler(async (req, res) => {
+  assert(req.body, PatchArticle);
+  const { aid } = req.params;
+  const aaid = parseInt(aid);
+  const updateDaTa = req.body;
+  const article = await prisma.article.update({
+    where: { id: aaid },
+    data: updateDaTa,
+  });
+
+  res.status(200).json(article);
+}));
+
+//////////////////////////////////////////////////////////////
+
+router.delete('/:aid', asyncHandler(async (req, res) => {
+    const { aid } = req.params;
+    const aaid = parseInt(aid);
+    await prisma.article.delete({
+      where: { id: aaid },
+    });
+    res.sendStatus(204);
+}));
+
+///////////////////////////////////////////////////////////
+
+// - 댓글 등록 API를 만들어 주세요.
+// - `content`를 입력하여 댓글을 등록합니다.
+// - 중고마켓, 자유게시판 댓글 등록 API를 따로 만들어 주세요.
+
+router.post('/:aid/comments', asyncHandler(async (req, res) => {
+  assert(req.body, CreateComment);
+  const { aid } = req.params;
+  const aaid = parseInt(aid);
+  const { content } = req.body;
+  const comment = await prisma.comment.create({
+    data: {
+      articleId: aaid,
+      content,
+    },
+  });
+  res.status(201).json(comment);
+}));
+
+/////////////////////////////////////////////////////////
+
+// - 댓글 목록 조회 API를 만들어 주세요.
+//     - `id`, `content`, `createdAt` 를 조회합니다.
+//     - cursor 방식의 페이지네이션 기능을 포함해 주세요.
+//     - 중고마켓, 자유게시판 댓글 목록 조회 API를 따로 만들어 주세요.
+
+router.get('/:aid/comments', asyncHandler(async (req, res) => {
+  const { aid } = req.params;
+  const { limit = 5, cursor } = req.query;
+  const aaid = parseInt(aid);
+  const comments = await prisma.comment.findMany({
```

Comment:
aaid 보다는 명확한 변수명인 articleId로 받아줘도 되지 않을까 싶습니다.
```suggestion
  const articleId = parseInt(aid);
```

Reply:
그리고 parseInt를 사용하는 모든 부분에서 리턴값 검증이 빠져있는 듯 합니다. 
parseInt는 NaN을 반환할 수 있는데 이 경우 적절한 에러 핸들링이 필요합니다.